docs/csharp/fundamentals/exceptions/snippets/exceptions/TaskExceptions.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace exceptions;
+
+class KitchenHelpers
+{
+    public class Toast { }
+
+    // <TaskExceptions>
+    // Non-async, task-returning method.
+    // Within this method (but outside of the local function),
+    // any thrown exceptions emerge synchronously.
+    public static Task<Toast> ToastBreadAsync(int slices, int toastTime)
+    {
+        if (slices is < 1 or > 4)
+        {
+            throw new ArgumentException(
+                "You must specify between 1 and 4 slices of bread.",
+                nameof(slices));
+        }
+
+        if (toastTime < 1)
+        {
+            throw new ArgumentException(
+                "Toast time is too short.", nameof(toastTime));
+        }
+
+        return ToastBreadAsyncCore(slices, toastTime);
+
+        // Local async function.
+        // Within this function, any thrown exceptions are stored in the task.
```

Comment:
I think this is fine, but I want to make a mention: comments aren't localized. I think this concept is covered well in the article text, so it's additional context. If you think something is missing in the article that non-English speakers will need to know, we should add a sentence or two in the article text.

Reply:
So it's a case of an unfair advantage for English readers I guess. Do you suggest I remove the comments, or add some language-agnostic markers in the code comments, or ...?

Reply:
I often have both. I want the comments because people copy and paste our samples. I also add text in the article so a reader whose not well versed in English can understand the full article.